ReactNativeClient/lib/TemplateUtils.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs-extra');
+const { time } = require('lib/time-utils.js');
+const Mustache = require('mustache');
+
+const TemplateUtils = {};
+
+// new template variables can be added here
+// If there are too many, this should be moved to a new file
+const view = {
+	date: time.formatMsToLocal(new Date().getTime(), time.dateFormat()),
+	time: time.formatMsToLocal(new Date().getTime(), time.timeFormat()),
+	datetime: time.formatMsToLocal(new Date().getTime()),
+	custom_datetime: () =>  { return (text, render) => {
+		return render(time.formatMsToLocal(new Date().getTime(), text));
+	}},
+}
+
+// Mustache escapes strings (including /) with the html code by default
+// This isn't useful for markdown so it's disabled
+Mustache.escape = (text) => { return text; }
+
+TemplateUtils.render = function(input) {
+	return Mustache.render(input, view);
+}
+
+TemplateUtils.loadTemplates = async function(filePath) {
+		let templates = [];
+		let filenames = [];
+
+		if (await fs.pathExists(filePath)) {
+			try {
+				filenames = fs.readdirSync(filePath);
```

Comment:
Please use shim.fsDriver().readDirStats (see fs-driver-node.js). In general async functions should be used to avoid blocking io, and shim.fsDriver should be used so that we can potentially use the same code in mobile too. Maybe not relevant in this particular case, but I prefer everything to go through fsDriver in general.

Reply:
Thanks for the info, I've updated all uses of fs to shim.fsDriver().